packages/@uppy/companion/src/server/helpers/utils.js

```diff
@@ -142,7 +142,16 @@ module.exports.decrypt = (encrypted, secret) => {
   return decrypted
 }
 
-module.exports.defaultGetKey = (req, filename) => `${crypto.randomUUID()}-${filename}`
+module.exports.defaultGetKey = (req, filename) => `${getS3Prefix()}${crypto.randomUUID()}-${filename}`
+
+/**
+ * Get the S3 prefix from the environment variable.
+ *
+ * @returns {string}
+ */
+function getS3Prefix () {
+  return `${process.env.COMPANION_S3_PREFIX}/ || ''`
```

Comment:
thanks for the pr! did you actually test this? 😅

```
`${undefined}/ || ''` becomes "undefined/ || ''"

`${'myprefix'}/ || ''` becomes "myprefix/ || ''"
```

on another note I think we shouldn't put env variables inside getS3Prefix because this function is also used by the non-standalone companion, which doesn't use process.env at all. I think the prefix should be an argument to `defaultGetKey`

Reply:
I'm not gonna lie. No I didn't test. It was a very quick attempt to get the ball rolling. Plus too many years as a manager means I am definitely not a Javascript developer!

Reply:
@mifi Had another crack